### PR TITLE
feat: Add support for BQ routine

### PR DESCRIPTION
**Pull request by akovtun-extenda**
This PR adds support for BQ Routines. More info: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_routine

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/README.md /tmp/tmp.7SAYMVThqi/generate_docs/workspace/README.md
195c195
< | routines | A list of objects which include table\_id, schema, clustering, time\_partitioning, range\_partitioning, expiration\_time and labels. | <pre>list(object({<br>    routine_id          = string,<br>    routine_type        = string,<br>    routine_language    = string,<br>    definition_body     = string,<br>    return_type         = string,<br>    routine_description = string,<br>    arguments = object({<br>      name            = string,<br>      data_type       = string,<br>      argument_kind   = string,<br>      mode            = string, <br>    }),<br>  }))</pre> | `[]` | no |
---
> | routines | A list of objects which include routine\_id, routine\_type, routine\_language, definition\_body, return\_type, routine\_description and arguments. | <pre>list(object({<br>    routine_id          = string,<br>    routine_type        = string,<br>    routine_language    = string,<br>    definition_body     = string,<br>    return_type         = string,<br>    routine_description = string,<br>    arguments = object({<br>      name            = string,<br>      data_type       = string,<br>      argument_kind   = string,<br>      mode            = string,<br>    }),<br>  }))</pre> | `[]` | no |
210c210
< | routine\_ids | Friendly names for any external tables being provisioned |
---
> | routine\_ids | Unique IDs for any routine being provisioned |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
variables.tf
--- old/variables.tf
+++ new/variables.tf
@@ -170,10 +170,10 @@
     return_type         = string,
     routine_description = string,
     arguments = object({
-      name            = string,
-      data_type       = string,
-      argument_kind   = string,
-      mode            = string,
+      name          = string,
+      data_type     = string,
+      argument_kind = string,
+      mode          = string,
     }),
   }))
 }
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: README.md [hunk 1]
```diff
@@ -150,6 +150,26 @@ The `views` variable should be provided as a list of object with the following k
   }
 }
 ```
+
+### Variable `routines` detailed description
+
+The `routines` variable should be provided as a list of object with the following keys:
+```hcl
+{
+  routine_id = "some_id"                      # The ID of the the routine. The ID must contain only letters, numbers, or underscores. The maximum length is 256 characters.
```
Comment (morgante) [inline README.md:159]: Prefixing these values with `return` is probably unnecessary.

File: README.md [hunk 2]
```diff
@@ -150,6 +150,26 @@ The `views` variable should be provided as a list of object with the following k
...
+  routine_type = "PROCEDURE"                  # The type of routine. Possible values are SCALAR_FUNCTION and PROCEDURE.
+  routine_language = "SQL"                    # The language of the routine. Possible values are SQL and JAVASCRIPT.
+  definition_body = "CREATE FUNCTION test return x*y;"  # The body of the routine. For functions, this is the expression in the AS clause. If language=SQL, it is the substring inside (but excluding) the parentheses.
+  return_type     = null                      # A JSON schema for the return type. Optional if language = "SQL"; required otherwise. If absent, the return type is inferred from definitionBody at query time in each query that references this routine. If present, then the evaluated result will be cast to the specified returned type at query time.
+  routine_description = "Description"         # The description of the routine if defined.
+  arguments = {                               # Set it to `null` to omit arguments block configuration for the routine.
```
Comment (morgante) [inline README.md:165]: Arguments should be an array of objects, right? Right now only one argument is allowed.

**Review by morgante — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (akovtun-extenda) [reply README.md:159]: do you mean variable `return_type`?

Comment (morgante) [reply README.md]: No I meant that they don't all need to be prefixed with `routine`, sorry for the typo.

Comment (akovtun-extenda) [reply README.md]: aah, okay. understood. I'll leave only `routine_id` and `routine_type` as that's how they are named in the actual resource. will remove it from other variables.

**Comment by akovtun-extenda** [comment]
@morgante Can you please take a look again?

**Review by morgante — APPROVED** [review]
Thanks for the addition!

**Pull request merged**